pytensor/compile/builders.py

```diff
@@ -400,6 +400,15 @@ def __init__(
             Check :func:`pytensor.function` for more arguments, only works when not
             inline.
         """
+        ignore_unused_inputs = kwargs.get("on_unused_input", False) == "ignore"
+        if len(inputs) != len(set(inputs)) and not ignore_unused_inputs:
```

Comment:
Tiny performance thing, do the second check before the len comp?

Reply:
it's fine to just switch the order inline right? The check will just stop going through the additional and conditions if the first condition fails?

Reply:
Yeah, python and/or/all/any are lazy, only evaluate when needed